From 6db723fdc3f7482efb5a914f12c2f858ae4897d7 Mon Sep 17 00:00:00 2001
From: Tyler Gunn <tgunn@google.com>
Date: Thu, 20 Sep 2018 14:56:19 -0700
Subject: [PATCH 15/39] Remap "SIP_NOT_FOUND" to INVALID_NUMBER.

Remapping CODE_SIP_NOT_FOUND error code to INVALID_NUMBER instead of
SERVER_UNREACHABLE.  Apparently SIP_NOT_FOUND is more analogous to
invalid number.

Bug: 111909355
Test: Added unit test
Merged-In: Ibe008c63b5d946013da19b8b9e7d4c9c312d050f
Change-Id: I4e927a565420fa215859fd2eee5b035a30db699f

# Conflicts:
#	tests/telephonytests/src/com/android/internal/telephony/imsphone/ImsPhoneCallTrackerTest.java
---
 .../internal/telephony/imsphone/ImsPhoneCallTracker.java  | 4 +++-
 .../telephony/imsphone/ImsPhoneCallTrackerTest.java       | 8 ++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java b/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java
index 7a5275c26..7127eba62 100644
--- a/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java
+++ b/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java
@@ -2006,10 +2006,12 @@ public class ImsPhoneCallTracker extends CallTracker implements ImsPullCall {
                 return DisconnectCause.SERVER_ERROR;
 
             case ImsReasonInfo.CODE_SIP_SERVICE_UNAVAILABLE:
-            case ImsReasonInfo.CODE_SIP_NOT_FOUND:
             case ImsReasonInfo.CODE_SIP_SERVER_ERROR:
                 return DisconnectCause.SERVER_UNREACHABLE;
 
+            case ImsReasonInfo.CODE_SIP_NOT_FOUND:
+                return DisconnectCause.INVALID_NUMBER;
+
             case ImsReasonInfo.CODE_LOCAL_NETWORK_ROAMING:
             case ImsReasonInfo.CODE_LOCAL_NETWORK_IP_CHANGED:
             case ImsReasonInfo.CODE_LOCAL_IMS_SERVICE_DOWN:
diff --git a/tests/telephonytests/src/com/android/internal/telephony/imsphone/ImsPhoneCallTrackerTest.java b/tests/telephonytests/src/com/android/internal/telephony/imsphone/ImsPhoneCallTrackerTest.java
index eac615927..ef9039998 100644
--- a/tests/telephonytests/src/com/android/internal/telephony/imsphone/ImsPhoneCallTrackerTest.java
+++ b/tests/telephonytests/src/com/android/internal/telephony/imsphone/ImsPhoneCallTrackerTest.java
@@ -812,4 +812,12 @@ public class ImsPhoneCallTrackerTest extends TelephonyTest {
         }
         verify(mImsPhone, times(1)).startOnHoldTone(nullable(Connection.class));
     }
+
+    @Test
+    @SmallTest
+    public void testSipNotFoundRemap() {
+        assertEquals(DisconnectCause.INVALID_NUMBER,
+                mCTUT.getDisconnectCauseFromReasonInfo(
+                        new ImsReasonInfo(ImsReasonInfo.CODE_SIP_NOT_FOUND, 0), Call.State.ACTIVE));
+    }
 }
-- 
2.17.1

